to0scheduler/to0serviceimpl/src/main/java/org/sdo/iotplatformsdk/to0scheduler/rest/RestClient.java

```diff
@@ -0,0 +1,205 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package org.sdo.iotplatformsdk.to0scheduler.rest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.UUID;
+
+import javax.net.ssl.SSLContext;
+
+import org.sdo.iotplatformsdk.common.rest.DeviceState;
+import org.sdo.iotplatformsdk.common.rest.SignatureResponse;
+import org.sdo.iotplatformsdk.to0scheduler.to0service.To0PropertiesLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+
+/**
+ * This class is responsible for sending the HTTPS request to Owner Companion
+ * Service.
+ */
+public class RestClient {
+
+  private final Logger logger = LoggerFactory.getLogger(RestClient.class);
+  private final SSLContext sslContext;
+  private final ObjectMapper objectMapper;
+
+  public RestClient(SSLContext sslContext, ObjectMapper objectMapper) {
+    this.sslContext = sslContext;
+    this.objectMapper = objectMapper;
+  }
+
+  private ObjectMapper objectMapper() {
+    return objectMapper;
+  }
+
+  private SSLContext sslContext() {
+    return sslContext;
+  }
+
+  /**
+   * Send a GET request to retrieve the owner voucher for the specified device
+   * identifier.
+   *
+   * @param deviceId the device identifier.
+   * @return the owner voucher.
+   */
+  public String getDeviceVoucher(final String deviceId) {
+    try {
+      final String apiServer = To0PropertiesLoader.getProperty("rest.api.server");
+      final String path = To0PropertiesLoader.getProperty("rest.api.voucher.path");
+      final String revisedPath = path.replace("{deviceId}", deviceId);
+      final URI uri = URI.create(apiServer + revisedPath);
+
+      final HttpClient httpClient = HttpClient.newBuilder().sslContext(sslContext())
+          .connectTimeout(Duration.ofSeconds(10)).build();
```

Comment:
Same comment as above for hard-coded values.

Reply:
Fixed by creating a class-level constant.